internal/linters/go/staticcheck/staticcheck.go

```diff
@@ -34,5 +38,43 @@ func staticcheckHandler(log *xlog.Logger, a linters.Agent) error {
 		a.LinterConfig.Args = append([]string{}, "-debug.no-compile-errors=true", "./...")
 	}
 
+	ctx := context.Background()
+	query := fmt.Sprintf("repo:%s/%s filename:go.mod github.com/zeromicro/go-zero",
+		a.PullRequestEvent.Repo.GetOwner().GetLogin(),
+		a.PullRequestEvent.Repo.GetName())
+	result, _, err := a.GithubClient.Search.Code(ctx, query, nil)
+	if err != nil {
+		log.Errorf("Error searching code: %v\n", err)
+	} else {
+		if *result.Total > 0 {
+			log.Info("add gozero filter")
+			a.OutputFilters = append(a.OutputFilters, &defaultGozeroFilter)
+		}
+	}
 	return linters.GeneralHandler(log, a, linters.GeneralParse)
 }
+
+var defaultGozeroFilter = GozeroOutputFilter{
+	regex:   `unknown JSON option "([a-z]+).*"`,
```

Comment:
This approach seems hardcoded. I'm wondering if `staticcheck` or the industry has any recommendations for this scenario?